spring-batch-infrastructure/src/test/java/org/springframework/batch/item/file/builder/FlatFileItemReaderBuilderTests.java

```diff
@@ -273,6 +273,51 @@ public void testComments() throws Exception {
 		assertNull(reader.read());
 	}
 
+	@Test
+	public void testEmptyComments() throws Exception {
```

Comment:
I agree on the fix, however this test does not test the case reported in BATCH-2837 because the input resource does not contain a line with the default comment prefix `#`.  Here is what I mean (I changed the target type to `String` since it is easier to test for this case):

```java
@Test
public void testEmptyComments() throws Exception {
	FlatFileItemReader<String> reader = new FlatFileItemReaderBuilder<String>()
			.name("itemReader")
			.resource(getResource("line1\n#line2"))
			.comments(new String[]{})
			.lineMapper(new PassThroughLineMapper())
			.build();

	reader.open(new ExecutionContext());
	String item = reader.read();
	assertEquals("line1", item);
	item = reader.read();
	assertEquals("#line2", item); // "#line2" should not be considered as comment
	assertNull(reader.read());
	reader.close();
}
```

Do you agree? BTW, this test passes with this PR.